JOSE_paper/paper.md

```diff
@@ -0,0 +1,254 @@
+This software is often written by individuals without a formal training in software engineering
+leading to inefficiencies, bugs, and poor reproducibility/re-usability.
+
+The course described in this paper is designed to raise awareness software
+engineering principles and how they can be applied to research code.
+It illustrates, through a series of practical exercises, why these are important and
+introduced basic techniques to produce FAIR-er code [@barker2022introducing].
+We hope that introducing a software engineering mindset, with simple tools and techniques
+to facilitate this will have long-term impact with participants continuing to explore
+ideas in their future work beyond this course.
+
+
+
+# Statement of need
+
+Software forms a significant part of research today, with much of it being written and developed
+by academic researchers. Many of these individuals are self-taught or have minimal training,
+which will often focus on 'coding' rather than software development.
+As a result there is often a lack of awareness of software development principles
+amongst academic developers leading to software that is hard to use and maintain,
+presenting a barrier to research.
+
+In our experience working with academic developers many of the bad practices and
+anti-patterns are a result of them:
+
+- learning on the job from old code and emulating habits they observe
+- never having been made aware of the available tools and techniques
+- not keeping up to date with language/package developments
+- not investing time into writing good code, often due to placing priorities elsewhere.
+
+The intent of developing this short course, _RSE Skills (in Python)_, is to educate
+individuals in some basic software engineering principles that they can carry forward
+in their work.
+It is designed to introduce key tools and techniques to facilitate the writing of
+better code, but also to raise awareness of research software engineering and provide pointers
+to allow researchers to continue exploring and learning beyond this material.
+
+
+
+# Learning objectives
+
+The key learning objective for the workshop is to _Introduce key tools and concepts of
+research software engineering, and how they can be applied in everyday use to write
+higher-quality code_.
+
+More specifically we cover:
+
+* Why software engineering principles matter
+* (Virtual) environments and dependencies
+* Code standards and best practice (through PEP)
+  * Code formatters and linting/static-analysis tools
+* Documentation
+  * Docstrings
+  * Advice on commenting
+  * The idea of self-documenting code and variable naming
+  * READMEs and other useful files in your repository
+  * Software licenses
+* General principles for better code
+  * Magic Numbers
+  * Removal of hard-coded content to configuration files
+  * use of f-strings in Python.
+
+
+
+# Teaching materials
+
+All of the teaching materials for this course are available online in a GitHub repository.
```

Comment:
Include link?

Reply:
Should be done on https://github.com/jatkinson1000/rse-skills-workshop/pull/14/commits/6f72db46956e9aea2f5480fea142a4c6dca3204b